extensions/blocks/map/map.php

```diff
@@ -25,7 +25,83 @@
 function jetpack_map_block_load_assets( $attr, $content ) {
 	$api_key = Jetpack_Options::get_option( 'mapbox_api_key' );
 
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		global $wp, $map_block_counter;
+		if ( ! $map_block_counter ) {
+			$map_block_counter = 0;
+		}
+		$iframe_url = home_url( $wp->request ) . '?map-block-counter=' . $map_block_counter;
```

Comment:
In order to account for a scenario where map blocks are appearing in an archive post loop as opposed to singular, I believe this should add the query arg to the permalink for the current post:

```suggestion
		$iframe_url = add_query_arg( 'map_block_counter', $map_block_counter[ get_the_ID() ], get_permalink() );
```

Also, this should then short-circuit if not `in_the_loop()`, no?

Reply:
Yes, good catch: https://github.com/Automattic/jetpack/pull/13405/commits/c15dfc0a47fc47b7fe1aa4b5537ee29ca59feb1e